src/group/managed_group/mod.rs

```diff
@@ -725,17 +782,23 @@ impl<'a> ManagedGroup<'a> {
     /// in the queue of pending proposals.
     pub fn self_update(
         &mut self,
+        key_store: &KeyStore,
         key_package_bundle_option: Option<KeyPackageBundle>,
     ) -> Result<(Vec<MLSMessage>, Option<Welcome>), ManagedGroupError> {
         if !self.active {
             return Err(ManagedGroupError::UseAfterEviction(UseAfterEviction::Error));
         }
 
+        let credential = self.credential()?;
```

Comment:
I see this block a lot in here. The code should be factored out somehow to avoid having so many copies of this code.

Reply:
I don't understand why this would be a problem for a helper function, but I haven't tried it myself. I was thinking about something along the lines of:

```
fn fetch_credential_bundle(credential: &Credential, key_store: &'a KeyStore) -> Result<CbGuard, ManagedGroupError> { 
    ... 
}
```

Reply:
Right, I don't think you can clean this up without changing `self.credential()` to return a reference.